pkg/repository/git.go

```diff
@@ -53,6 +55,10 @@ func newGitRepoController(path string) gitRepoController {
 		repoGetter: gitRepositoryGetter{},
 		path:       path,
 		fs:         afero.NewOsFs(),
+		ignorer: ignore.NewMultiIgnorer(
+			ignore.NewDockerIgnorer(".dockerignore"),
+			ignore.NewOktetoIgnorer(".oktetoignore").BuildOnly(),
+		),
```

Comment:
Would it make sense to add an environment variable to disable this behavior in case something bad happens and builds are not re-triggered when needed? By default it would have this behavior, but if some customer is affected by something unexpected we can tell them to set that env and don't do any ignore (it would be just to create here a never ignorer or a multi ignore). WDYT?

Reply:
As discussed offline, we will add a feature flag to enable this behavior. By default, it will be disabled. 

I was thinking about the env var name. What do you think about any of these:
* `OKTETO_SMART_BUILDS_IGNORE_FILES_ENABLED`
* `OKTETO_SMART_BUILDS_USE_IGNORE_FILES`
* `OKTETO_SMART_BUILDS_READ_IGNORE_FILES`